core/shared/src/main/scala/org/http4s/headers/X-Content-Type-Options.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package headers
+
+import cats.data.NonEmptyList
+import cats.parse.Parser
+import org.http4s.internal.parsing.CommonRules
+import org.typelevel.ci.CIStringSyntax
+
+sealed trait `X-Content-Type-Options`
+
+object `X-Content-Type-Options`
+    extends HeaderCompanion[`X-Content-Type-Options`]("X-Content-Type-Options") {
+
+  case object nosniff extends `X-Content-Type-Options`
```

Comment:
Have only been half following the discussion, but here is my 2 cents:
1. It should be an `abstract class X-Content-Type-Options private {}`. This way it's not `sealed`, so we can add new members, but nobody else.
2. it should be `val nosniff: X-Content-Type-Options`, no reason to leak a type there plus it will be annoying for type inference.

Reply:
I am personally 👍 for p.2 – this is similar to what I'm thinking (see my comment above). Still not sure if it is really necessary though, but just in case.

In regards to p.1. Do you mean that in a case of sealed trait or abstract class adding more inheritors will break the binary compatibility? Even though we do not expose their types to the public scope...

Reply:
> Even though we do not expose their types to the public scope...

Great point. If all the member types are private, then it is probably ok for it to be sealed.

Still, I think an `abstract class` is preferable to a `trait` (this is how Scala 3 encodes enums). But it's not a big deal.